### DISCUSSION TITLE
Preventing Automatic Update of List When Fetching Specific Record

### DISCUSSION THREAD

**Discussion by chemseddine201** [TanStack/query#5677] [General]
I have two queries with different keys: ['signup-list'] to retrieve a list of records, and ['signup-list', 'id'] to fetch a single record. However, when I use the '**useQuery**' function to fetch a specific record, the corresponding record in the list automatically updates. I would like to know how to prevent this automatic update of the record in the list until I explicitly refresh the list.

**Comment by samhirtarif** [comment]
Dose the `queryFn` for the `queryKey` `['signup-list']` get executed?

**Reply by chemseddine201** [reply]
no the queryFn not executed but the data of item in the list in the query cache updated.

**Reply by samhirtarif** [reply]
Would it be possible for you to share a code snippet?

**Reply by chemseddine201** [reply]
single item fetch:
const query = useQuery(['umm-signup', id], () => baseApi.get(`/api/signup-list/${id}`), {
    enabled: !!id && (edit || duplicate),
    keepPreviousData: true,
    retry: false,
    refetchOnReconnect: true,
    refetchOnWindowFocus: false
  });
  
 list of items fetch: 
  const { status, isLoading, isError, isSuccess, data } = useQuery(
    ['umm-signup-list', state], //state is an object of filters
    () => baseApi.get(`/api/signup-list?${queryStr}`),
    {
      onError: (err) => {
        //reload();
      }
    }
  );

**Reply by samhirtarif** [reply]
Are these `useQuery` usages in the same component?
If yes, it would be better if you could share the code for the entire component. And please do add it as a code block, makes it much easier to read

**Reply by chemseddine201** [reply]
The useQuery usages are not within the same component. The main issue is that when fetching data, the item gets updated in the cached data list, which is unintended.

**Reply by samhirtarif** [reply]
I doubt that the item in the list can get updated without the list `useQuery` being called again. So I am trying to figure out if there is a way that its gets called again for some reason.
Is the `state` value passed in the list `useQuery` key `['umm-signup-list', state]` local to that component?

It would really help if you could share the code for both components that have these `useQuery` instances

**Reply by chemseddine201** [reply]
The problem caused by a wrong implementation in the following function: 

`export const initialValues = (query, duplicate=false, uniqueFields=[], itemFields=[], data = {}) => {
  let resultData = data;
  if (query?.isSuccess) {
    const item = query?.data?.data?.data?.item;
    if (item && Object.keys(item).length) {
      resultData = {...resultData, ...item};
    }
  }`

The error occurred because the variable resultData was holding a reference to the object data instead of its value. To resolve this issue, i assign a copy of the data object to resultData, like this: resultData = {...data}. This ensures that any updates made to resultData won't affect the original reference and the cached data in the list with the query key ["umm-signup-list"].
